build/azure-pipelines/sql-product-build.yml

```diff
@@ -129,7 +129,20 @@ stages:
     #   timeoutInMinutes: 90
 
   - stage: Release
-    condition: and(succeeded(), or(eq(variables['VSCODE_RELEASE'], 'true'), and(eq(variables['VSCODE_QUALITY'], 'insider'), eq(variables['Build.Reason'], 'Schedule'))))
+    # Requirements:
+    # 1. Release can be created only from builds of the official build pipeline, but not all builds of it should be released automatically.
+    # 2. A build should be released only when the release variable is set to true.
+    # 3. The daily scheduled Insiders build should be released automatically.
+    #
+    # Limitation:
+    # Ideally, we should be able to use the condition `and(succeeded(), eq(variables['VSCODE_RELEASE'], 'true')` to achieve what we want. The VSCODE_RELEASE variable can be defined as overwritable or non-overwritable based on whether we want to
+    # release from the pipeline. Unfortunately ADO doesn't allow the use of variables for scheduled job as per:
+    # https://learn.microsoft.com/en-us/azure/devops/pipelines/process/scheduled-triggers?view=azure-devops&tabs=yaml#scheduled-triggers, as a result, we cannot set the VSCODE_RELEASE variable to true for scheduled Insider builds.
```

Comment:
I'm confused by this - what do you mean "doesn't allow the use of variables for scheduled jobs"? Variables definitely are used - and from what I can tell that isn't the issue here. The issue is that the original check was 

VSCODE_RELEASE==true OR (VSCODE_QUALITY==insiders && Build.Reason==Scheduled)

So while VSCODE_RELEASE was indeed set to false it was the second set of comparisons that caused it to still run the release stage. 

I'm not really sure we need a whole new variable - instead why not just make VSCODE_RELEASE do what it says and require that for ALL releases. So the new logic would be 

VSCODE_RELEASE==true && (VSCODE_QUALITY!=Stable OR (VSCODE_QUALITY==Insiders && Build.Reason==Schedule))

Reply:
doesn't allow the use of variables in the schedule information means you cannot specify for this scheduled build, use vscode_release = true. that's why we had the condition as: vscode_release = true or (insiderd && scheduled)

Reply:
But VSCODE_QUALITY is a variable too...

Reply:
(I realize the docs say `You can't use pipeline variables when specifying schedules.` but I honestly don't know what they mean by that since it doesn't appear to be true given that our pipelines are running just fine on a schedule with many variables defined)

Reply:
yes, but insider is the default value. you can't have a scheduled run to produce stable builds

Reply:
as I mentioned in the comment, if we can specify values for variables, all that we need is: `and(succeeded(), eq(variables['VSCODE_RELEASE'], 'true')`

Reply:
Ok, so what you're saying is that "you can't have the schedules for a pipeline have different variable values than manual/other types of runs"? Because yes, that seems accurate as far as I know but also I don't know why we need that level of granularity. VSCODE_RELEASE should only ever be true on a the product build pipeline. 

The reason for the additional checks is just safety - we don't want someone to accidently kick off a manual run of the product build pipeline and have that cause an insiders build to be released. (I think the build reason for re-runs of failed scheduled builds should still be Schedule so we can get insiders builds out by re-running if they fail for some reason).

Reply:
updated the PR according to our offline discussion.